jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/actions/ViewResourceAction.kt

```diff
@@ -0,0 +1,22 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.explorer.actions
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.project.DumbAware
+import software.aws.toolkits.jetbrains.core.explorer.ViewResourceDialog
+import software.aws.toolkits.jetbrains.core.explorer.nodes.AwsExplorerNode
+
+abstract class ViewResourceAction<in T : AwsExplorerNode<*>>(private val actionTitle: String, val resourceType: String, private val prefix: String) :
+    SingleExplorerNodeAction<T>(actionTitle), DumbAware {
```

Comment:
view resource implies `SingleResourceNodeAction`. is that what you intended?

Reply:
The action is based off the S3 service node not the buckets which is why used` SingleExplorerNodeAction` instead of `SingleResourceNodeAction`, but the action is to view a specific resource(here: bucket)